### PR TITLE
Feature/8689je9cp/list-startup-application

### PR DESCRIPTION
**Pull request by CristhianRF24**
I concluded the List Startup application  user story

### CODE REVIEW

**Comment by BelenA25** [comment]
You are making the pull request to main and not to the corresponding branch

**Review by MartinArispeRiveros — APPROVED** [review]

**Review by BelenA25 — APPROVED** [review]

File: src/app/startups/applications/addApplications/page.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+"use client";
```
Comment (richi0085) [inline src/app/startups/applications/addApplications/page.tsx]: Please move this logic to 
pages-components folder 
and here just import the component

File: src/app/startups/applications/addApplications/page.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    startup_name: string;
+    full_name: string;
+    email: string;
+    phone: string; 
+    city: string;
+    startup_description: string;
+    startup_stage: string;
+  }) => {
+    try {
+      const newApplicationRef = ref(database, `applications/${Date.now()}`);
```
Comment (richi0085) [inline src/app/startups/applications/addApplications/page.tsx]: move this to applications service

File: src/app/startups/applications/page.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
+"use client";
```
Comment (richi0085) [inline src/app/startups/applications/page.tsx:1]: move all logic to  page-components

File: src/app/startups/applications/page.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    window.addEventListener("resize", handleResize);
+    handleResize();
+
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  }, []);
+
+  const fetchInitialData = async () => {
+    const applicationsRef = ref(database, TYPE);
```
Comment (richi0085) [inline src/app/startups/applications/page.tsx]: move this logic to a service please

File: src/app/startups/applications/page.tsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    let queryApplications;
+
+    if (currentPage === 1) {
```
Comment (richi0085) [inline src/app/startups/applications/page.tsx]: 1 is a magic number, please create a Cosnt

File: src/app/startups/applications/page.tsx [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+  const handleSearch = (searchTerm: string) => {
+    if (!searchTerm.trim()) {
+      setIsSearching(false);
+      setCurrentPage(1); // Set to the first page
+      fetchInitialData(); // Fetch initial data if search term is empty
+      return;
+    }
+    setIsSearching(true);
+    const applicationsRef = ref(database, TYPE);
+    const searchQuery = query(
```
Comment (richi0085) [inline src/app/startups/applications/page.tsx]: move all this to a service

File: src/app/startups/page.tsx [hunk 7]
```diff
@@ -6,36 +6,89 @@ import TableSection from "@/components/table-section";
 import Title from "@/components/title";
```
Comment (richi0085) [inline src/app/startups/page.tsx:6]: move all logic to
page-components/startupsPage.tsx

File: src/app/startups/page.tsx [hunk 8]
```diff
@@ -6,36 +6,89 @@ import TableSection from "@/components/table-section";
...
 import { database } from "../firebaseConfig";
-import { onValue, ref } from "firebase/database";
-
-const ITEMS_PAGE = 6;
+import { limitToFirst, onValue, orderByKey, query, ref, startAfter, startAt } from "firebase/database";
 
 export default function Startups() {
     const [currentPage, setCurrentPage] = useState(1);
     const [data, setData] = useState<any[]>([]);
+    const [itemsPerPage, setItemsPerPage] = useState(6);
```
Comment (richi0085) [inline src/app/startups/page.tsx:14]: 6 is a magic number

File: src/app/startups/page.tsx [hunk 9]
```diff
@@ -6,36 +6,89 @@ import TableSection from "@/components/table-section";
...
+        return () => {
+            window.removeEventListener('resize', handleResize);
+        };
     }, []);
 
-    const totalPages = Math.ceil(data.length / ITEMS_PAGE);
-    const paginatedData = data.slice((currentPage - 1) * ITEMS_PAGE, currentPage * ITEMS_PAGE);
+    useEffect(() => {
+        const fetchingData = async () => {
+            const startupsReference = ref(database, 'startups');
```
Comment (richi0085) [inline src/app/startups/page.tsx:36]: move firebaselogic to a service

File: src/app/startups/page.tsx [hunk 10]
```diff
@@ -6,36 +6,89 @@ import TableSection from "@/components/table-section";
...
+            let queryStartups;
+
+            if (currentPage === 1) {
```
Comment (richi0085) [inline src/app/startups/page.tsx:39]: 1 is a magic number

File: src/app/startups/page.tsx [hunk 11]
```diff
@@ -6,36 +6,89 @@ import TableSection from "@/components/table-section";
...
+                );
+            } else {
+                queryStartups = query(
+                    startupsReference,
+                    orderByKey(),
+                    startAfter(lastKey),
+                    limitToFirst(itemsPerPage)
+                );
+            }
+            onValue(queryStartups, (snapshot) => {
```
Comment (richi0085) [inline src/app/startups/page.tsx:53]: move all firebase logic to a service

File: src/components/application-form.tsx [hunk 12]
```diff
@@ -0,0 +1,229 @@
+import { FormEvent, useState } from "react";
+import { z } from "zod";
+import { Button } from "./ui/button";
+import { Input } from "./ui/input";
+import { Label } from "./ui/label";
+import { Textarea } from "./ui/textarea";
+
+// Definimos el esquema de validación con Zod
+const applicationSchema = z.object({
```
Comment (richi0085) [inline src/components/application-form.tsx]: move this schema to shared/api/validation 
please

File: src/components/application-form.tsx [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+
+    // Validate forms data
+    const result = applicationSchema.safeParse(applicationData);
+
+    if (!result.success) {
+      setErrors(result.error);
+      return;
+    }
+
+    onSubmit(applicationData);
```
Comment (richi0085) [inline src/components/application-form.tsx]: move all this to a clear method

File: src/components/side-menu.tsx [hunk 14]
```diff
@@ -1,89 +1,173 @@
...
+          className={`flex-grow w-56 relative transition-all  h-[calc(100vh-2rem)] bg-[#FFCDC2] rounded-lg p-4 my-4 ml-4 flex-shrink-0 transition-transform duration-300 ease-in-out z-10`}
+        >
+          <NavigationMenu>
+            <NavigationMenuList className="flex flex-col space-y-6 ">
+              <NavigationMenuItem>
+                <Image
+                  src="https://i.ibb.co/1LZ6zNV/image-2-removebg-preview.png"
+                  width={125}
+                  height={100}
+                  alt=""
```
Comment (richi0085) [inline src/components/side-menu.tsx]: alt is important

File: src/components/side-menu.tsx [hunk 15]
```diff
@@ -1,89 +1,173 @@
...
+                    className={`w-full py-2 p-7 rounded-md hover:bg-[#F2A594] transition-colors ${getActiveClass(
+                      `/${item.toLowerCase()}`
+                    )}`}
+                  >
+                    <a
+                      href={`${item.toLowerCase()}`}
+                      className="flex space-x-4"
+                    >
+                      <span>
+                        {item === "Fundadores" && (
```
Comment (richi0085) [inline src/components/side-menu.tsx]: Fundadores is a magic strinc, create an enum if necesary

**Review by richi0085 — CHANGES_REQUESTED** [review]

File: src/components/addApplications.tsx [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+    startup_name: string;
+    full_name: string;
+    email: string;
+    phone: string;
+    city: string;
+    startup_description: string;
+    startup_stage: string;
+  }) => {
+    try {
+      const applicationsRef = ref(database, "applications");
```
Comment (richi0085) [inline src/components/addApplications.tsx]: Could you create a service please 
an applicationsService.ts

and from locations you should just call the service
applicationsService.get (or, update, or delete) 

if you dont do it we will have a lot of repited code on the project

File: src/components/pagination-section.tsx [hunk 17]
```diff
@@ -1,39 +1,110 @@
...
+      onPageChange(page);
+    }
+  };
+
+  
+  const getPageNumbers = () => {
+    let pages = [];
+    let startPage: number, endPage: number;
+
+    if (totalPages <= PAGE_RANGE * 2 + 1) {
```
Comment (richi0085) [inline src/components/pagination-section.tsx]: what is *2 + 1 
Please lets not have magic numbers

**Review by richi0085 — CHANGES_REQUESTED** [review]

**Review by MartinArispeRiveros — APPROVED** [review]

File: src/app/startups/applications/create/page.tsx [hunk 18]
```diff
@@ -0,0 +1,11 @@
+"use client";
+
+import AddApplications from "@/components/addApplications";
+
+export default function Create() {
+  return (
+    <div className="container mx-auto p-4">
```
Comment (richi0085) [inline src/app/startups/applications/create/page.tsx]: If a lot of views will have this class it shoul be a component called Layout or something like that

File: src/components/search-card.tsx [hunk 19]
```diff
@@ -2,13 +2,33 @@ import { Input } from './ui/input';
...
+        const value = e.target.value;
+        setSearchTerm(value);
+        onSearchClick(value);
+    };
     return (
         <Card className="mx-8 p-4 flex items-center space-x-5 border-none">
             <Typography tag='h6' noWrap={true}>Buscar Startup: </Typography>
-            <Input type="text" placeholder="Buscar..." className="flex-grow p-2 border border-gray-300 rounded-l-lg focus:outline-none focus:ring-2 focus:ring-green-500" />
-            <SearchButton></SearchButton>
+            <Input type="text" value={searchTerm} onChange={handleSearchChange}  placeholder="Escribe el nombre del startup..." onKeyDown={handleKeyDown} className="flex-grow p-2 border border-gray-300 rounded-l-lg focus:outline-none focus:ring-2 focus:ring-green-500" />
```
Comment (richi0085) [inline src/components/search-card.tsx]: to large 
please do something like this:
		<Input
			type="text"
			value="{searchTerm}"
			onChange="{handleSearchChange}"
			placeholder="Escribe el nombre del startup..."
			onKeyDown="{handleKeyDown}"
			className="flex-grow p-2 border border-gray-300 rounded-l-lg focus:outline-none focus:ring-2 
                        focus:ring-green-500"
		/>

File: src/components/side-menu.tsx (hunk 15 above)
Comment (richi0085) [reply src/components/side-menu.tsx]: solve it please

**Review by richi0085 — CHANGES_REQUESTED** [review]

**Review by MartinArispeRiveros — APPROVED** [review]

**Review by richi0085 — APPROVED** [review]

**Pull request merged**